test_node.py

```diff
@@ -1,19 +1,133 @@
-    assert "v{}".format(container.version) in container.connection.check_output(
+    assert f"v{container.version}" in container.connection.check_output(
         "node -v"
     )
 
 
 # We don't care about the version, just test that the command seem to work
 def test_npm(container):
     assert container.connection.run_expect([0], "npm version")
+    assert container.connection.run_expect([0], "yarn --version")
+
+
+@pytest.mark.parametrize(
+    "npm_package",
+    [
+        pkg.to_pytest_param()
+        for pkg in (
+            NpmPackageTest(
+                repository_url="https://github.com/lodash/lodash.git",
+                build_command="npm ci && npm test",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/chalk/chalk.git",
+                build_command="npm install && npm test",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/tj/commander.js.git",
+                build_command="npm ci && npm test && npm run lint",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/visionmedia/debug.git",
+                build_command="npm install && npm run lint && npm run test:node",
+            ),
+            NpmPackageTest(
+                repository_url="https://github.com/expressjs/express.git",
+                build_command="""npm config set shrinkwrap false &&
+            npm rm --silent --save-dev connect-redis &&
+            npm run test-ci &&
+            npm run lint
+""",
+            ),
+            NpmPackageTest(
+                build_command="""npm install -g grunt-cli &&
+            npm install &&
+            grunt test
+""",
+                repository_url="https://github.com/moment/moment",
+            ),
+            NpmPackageTest(
+                build_command="npm ci && npm test",
+                repository_url="https://github.com/caolan/async",
+            ),
+            NpmPackageTest(
+                build_command="yarn --frozen-lockfile && yarn test",
+                repository_url="https://github.com/facebook/react.git",
+            ),
+        )
+    ]
+    + [
+        pkg.to_pytest_param(marks=mark)
+        for pkg, mark in (
+            (
+                NpmPackageTest(
+                    build_command="""yarn install &&
+                npm install --no-save "react@~16" "react-dom@~16" &&
+                yarn run pretest &&
+                yarn run tests-only &&
+                yarn run build
+""",
+                    repository_url="https://github.com/facebook/prop-types",
+                ),
+                pytest.mark.skip(reason="Broken for some reason"),
+            ),
+            (
+                NpmPackageTest(
+                    repository_url="https://github.com/jprichardson/node-fs-extra",
+                    build_command="npm install && npm run unit",
+                ),
+                pytest.mark.xfail(
```

Comment:
nit: I can't shake why I would do it, but I would pass the marks as kwargs of NpmPackageTest, and just do a single list comprehension for all the tests `[ pkg.to_pytest_param(marks=self.marks) for pkg in (NpmPackageTest(build_command=, repository_url=, marks=` marks being only set  on L103+ . WDYT? Will it be more readable for non pythonistas?

Reply:
Yeah, originally I wanted to treat `NpmPackageTest` as a pure data container, but now that it also contains some "business logic", I think adding the mark there makes sense as well.